ClearKeep/Scenes/Search/Views/SearchContentView.swift

```diff
@@ -0,0 +1,254 @@
+	@State private(set) var searchCatalogy: SearchCatalogy = .all
+	@Binding var imageUser: Image
+	@Binding var userName: String
+	@Binding var message: String
+	@Binding var groupText: String
+	@Binding var dateMessage: String
+
+	// MARK: - Init
+	init(searchCatalogy: SearchCatalogy = .all,
+		 imageUser: Binding<Image>,
+		 userName: Binding<String>,
+		 message: Binding<String>,
+		 groupText: Binding<String>,
+		 dateMessage: Binding<String>) {
+		self._searchCatalogy = .init(initialValue: searchCatalogy)
+		self._imageUser = imageUser
+		self._userName = userName
+		self._message = message
+		self._groupText = groupText
+		self._dateMessage = dateMessage
+	}
+	
+	// MARK: - Body
+	var body: some View {
+		content
+			.background(backgroundColorView)
+			.edgesIgnoringSafeArea(.all)
+	}
+}
+
+// MARK: - Private
+private extension SearchContentView {
+	var content: AnyView {
+		AnyView(contentView)
+	}
+
+	var searchCatalogContent: AnyView {
+		AnyView(searchCatalogView)
+	}
+
+	var allContent: AnyView {
+		AnyView(allView)
+	}
+
+	var peopleContent: AnyView {
+		AnyView((peopleView))
+	}
+
+	var groupContent: AnyView {
+		AnyView(groupView)
+	}
+
+	var messageContent: AnyView {
+		AnyView(messageView)
+	}
+}
+
+// MARK: - Private variable
+private extension SearchContentView {
+	var backgroundColorView: Color {
+		colorScheme == .light ? AppTheme.shared.colorSet.background : AppTheme.shared.colorSet.black
+	}
+	
+	var foregroundColorButton: Color {
+		colorScheme == .light ? AppTheme.shared.colorSet.primaryDefault : AppTheme.shared.colorSet.offWhite
+	}
+
+	var foregroundColorTitle: Color {
+		colorScheme == .light ? AppTheme.shared.colorSet.grey1 : AppTheme.shared.colorSet.greyLight
+	}
+
+	var backgroundButton: LinearGradient {
+		colorScheme == .light ? backgroundButtonLight : backgroundButtonDark
+	}
+
+	var backgroundButtonDark: LinearGradient {
+		LinearGradient(gradient: Gradient(colors: AppTheme.shared.colorSet.gradientPrimary), startPoint: .leading, endPoint: .trailing)
+	}
+
+	var backgroundButtonLight: LinearGradient {
+		LinearGradient(gradient: Gradient(colors: [AppTheme.shared.colorSet.offWhite, AppTheme.shared.colorSet.offWhite]), startPoint: .leading, endPoint: .trailing)
+	}
+}
+
+// MARK: - Private
+private extension SearchContentView {
+	func allAction() {
+		self.searchCatalogy = .all
+
+	}
+
+	func peopleAction() {
+		self.searchCatalogy = .people
+
+	}
+
+	func groupAction() {
+		self.searchCatalogy = .group
+
+	}
+
+	func messageAction() {
+		self.searchCatalogy = .message
+
+	}
+}
+
+// MARK: - Loading Content
+private extension SearchContentView {
+	var contentView: some View {
+		VStack(alignment: .leading, spacing: Constants.spacingContent) {
+			searchCatalogContent
+			HStack {
+				Group {
+					switch searchCatalogy {
+					case .all:
+						allContent
+					case .people:
+						peopleContent
+					case .group:
+						groupContent
+					case .message:
+						messageContent
+					}
+				}
+			}
```

Comment:
Make a list view horizontal. Maybe future will have more options.
Create a subview for this.